### DISCUSSION TITLE
Integrated AI Markdown Proofing and Post Production

### DISCUSSION THREAD

**Discussion by phaynes** [zed-industries/zed#23912] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

>  From spell check #5305  Future AI ideas

> A distant future goal might be to somehow leverage the supported AI models in Zed to fill suggestions for mispelled
 > words, if the suggestions provided by some crate aren't the greatest.

When Markdown editing I frequently switch between Zed, ChatGPT, Gemini, etc., with three main use cases:

1. *Proofing*: My standard response settings are tailored for specific styles, as exemplified below. I need the proofreading to adhere to a distinct style based on the context—basic questions, formal responses, and technical discourse—and to be delivered in markdown format using British English spelling.
So for example, part of my chatGPT response template is -
"Basic questions should be concise and direct, emulating the style of the Australian Daily Telegraph, without colloquialisms. Formal responses should be brief and direct, utilising a military writing style. Technical responses should cater to a postgraduate audience and remain technical."
Ideally, I'd like the option to easily select the level of proofing required right after highlighting the text I've just written. i.e a "quick proof", followed by a more reasoned proof, where I give the AI's more context to assist in their feedback.

2. *Argument Improvement & Revision:* I alternate between AI tools to refine and enhance my work. Direct integration of such feedback within Zed would streamline my editing process.

3. *Post-Production:* The final adjustments include ensuring APA v7 compliance for references, adding document headers, and formatting the output to specific standards (PDF/HTML or voice script).

For this to be done, I need a settings location to put my editing prompts, a key stroke to initiate it. When the key stroke is executed, someway to trivially see the differences and then accept, reject or edit, prior to replacing the original text.

Incorporating these features into Zed would significantly enhance my editing workflow, minimising the need to switch between different tools for each task, and negate the need for inline spell checks (which I have never  seen done well in a programming environment).

**Comment by maxdeviant** [comment]
Some interesting thoughts here! I'm going to move to a discussion to discuss further.
